_sips/minutes/2020-03-12-minutes.md

```diff
@@ -0,0 +1,166 @@
+* Martin: that's because the docs target users
+* Gui: but for the SIP process we should
+* Gui: also users ask "why can't we just 'fix' the existing stuff?"
+* Gui: why instead of a big redesign let's just, for example, fix mulit-parameter implicit classes?
+* Martin: we need a fresh start that fixes the 6 different aspects systematically
+* Martin: some of the existing problems/bugs are just not fixable
+* Gui: people are uneasy with not knowing how things desugar
+* Martin: I still think this is within the bandwidth of what Scala's always done
+* Gui: another concern is for anonymous things, the compiler has to come up with a name, should we specify that scheme?
+* Seb: it's too fragile, for instance if you rename a type, the name changes, or a type parameter name, or whether it's fully qualified or partially imported
+* Martin: and the names are optimised to be short, and not optimised not to clash
+* Seb: we should highlight that well in the docs: library authors, name your instances (and your method extension given instances)
+
+### Review given imports
+
+* Gui: one inconsistency is that doing a named import of a given is without the `given` keyword: `import A.foo`
+* Gui: and IDEs will want to import the most specific thing, so the synthetic name of anonymous instances, particularly extension names
+* Gui/Seb: we could add `extension` to the import syntax
+* Dale: the previous discussion about libraries always wanting to name their given instances and extensions, most things will have good names
+* Dale: so if IDEs add imports with synthetic names in apps, doesn't seem to really matter
+
+### Extension methods
+
+* Gui: one difference between infix extension methods and grouped extension methods is that grouped they live in a given instance
+* Martin: people like the `extension` keyword, when we have it as only `given` people were confused by what these things were
+* Gui: one big difference with the status quo (Scala 2) of extension methods is you can't have two sets of type parameters, on the class and on the def, and people aren't pleased with that
+* Martin: eventually that will go when we have curried type parameters, which isn't a 3.0 thing
+* Gui: could we add `private` support to extension methods?
+* Martin: Actually I just tried it, we do allow private
+
+### Export clauses
+
+* Martin: addressing the problem that Scala don't assist encouraging composition (over inheritance)
+* Martin: side reasons are package object inhertance and exporting enum cases
+* Martin: the feedback is some want it to do more, some want it to do less
+* Martin: I feel good about how it is now, I wouldn't change any detail on it
+* Adriaan: it would be good to add a type ascription when exporting (`export A.T: Foo` to only export Foo's methods)
+* Martin: that wouldn't be big, we could add that
+* Gui: we could add refinements to the import, to select exactly the method used (`import A.T { def foo(x: Int): Int }`)
```

Comment:
I'm pretty sure @smarter's comment here was about exports, not imports. So something like `export A.T: { def foo(x: Int): Int }`.

Reply:
Also the syntax is:
```scala
export (A: Foo).T // already mentioned in https://contributors.scala-lang.org/t/request-for-comments-on-exports/4051/17
export (A: { def foo(x: Int): Int }).foo
```
You put the ascription on the value who's being upcasted, just like in regular code. 
I think we also briefly touched on the fact that once we allow these sort of ascriptions for export, we could allow them on imports for symmetry, but that's not a very important point.